datafusion/physical-plan/src/filter.rs

```diff
@@ -345,19 +360,55 @@ struct FilterExecStream {
     input: SendableRecordBatchStream,
     /// runtime metrics recording
     baseline_metrics: BaselineMetrics,
+    /// Whether to allow an input batch to be returned unmodified in the case where
+    /// the predicate evaluates to true for all rows in the batch
+    reuse_input_batches: bool,
 }
 
 pub(crate) fn batch_filter(
     batch: &RecordBatch,
     predicate: &Arc<dyn PhysicalExpr>,
+    reuse_input_batches: bool,
 ) -> Result<RecordBatch> {
     predicate
         .evaluate(batch)
         .and_then(|v| v.into_array(batch.num_rows()))
         .and_then(|array| {
             Ok(match as_boolean_array(&array) {
                 // apply filter array to record batch
-                Ok(filter_array) => filter_record_batch(batch, filter_array)?,
+                Ok(filter_array) => {
+                    if reuse_input_batches {
+                        filter_record_batch(batch, filter_array)?
+                    } else {
+                        if filter_array.true_count() == batch.num_rows() {
+                            // special case where we just make an exact copy
```

Comment:
I think I am missing something -- why go through all the effort with `MutableArrayData` 

In other words, why isn't thus simply
```rust
Ok(batch.clone())
```

To literally return the input batch?

I think that would be less code and faster

Reply:
That would be the same that the filter.

The use case as far as I understand is that for comet the data needs to be a new copy, as spark will reuse the existing data/arrays.

Reply:
See https://github.com/apache/datafusion-comet/pull/835

Reply:
I see -- I think it would help if we made a function with a name that made it clear what was going on, something like

```rust
fn force_new_data_copy(...)
```

I also left a suggestion here